privacyidea/lib/eventhandler/requestmangler.py

```diff
@@ -0,0 +1,163 @@
+        
+        At the time of writing I can not make up a scenario, but technically it could 
+        make sense.
+        """
+        return ["post", "pre"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        actions = {ACTION_TYPE.DELETE:
+                       {"parameter":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The parameter that should be deleted.")}
+                        },
+                   ACTION_TYPE.ADD:
+                       {"parameter":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The parameter that should be added or modified.")
+                             },
+                        "value":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The new value of the parameter. Can contain tags like {0}, {1} for  "
+                                              "the matched sub strings.")
+                            },
+                        "match_parameter":
+                            {"type": "str",
+                             "description": _("The parameter, that should match some values.")
+                            },
+                        "match_pattern":
+                            {"type": "str",
+                             "description": _("The value of the match_parameter. It can contain a regular "
+                                              "expression and '()' to transfer values to the new parameter.")
+                            }
+                        }
+                   }
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action:
+        :param options: Contains the flask parameters g, request, response
+            and the handler_def configuration
+        :type options: dict
+        :return:
+        """
+        ret = True
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options", {})
+        parameter = handler_options.get("parameter")
+
+        if parameter:
+            if action.lower() == ACTION_TYPE.DELETE:
+                if parameter in request.all_data:
+                    del(request.all_data[parameter])
+            elif action.lower() == ACTION_TYPE.ADD:
+                value = handler_options.get("value")
+                match_parameter = handler_options.get("match_parameter")
+                match_pattern = handler_options.get("match_pattern")
+                if value:
```

Comment:
This wouldn't allow us to add a parameter whose value is the empty string. Do you think we would ever need this?

Reply:
I thought about this, if a value should have an empty parameter. Probably not. So you can delete the parameter.
I am not sure, if there are any cases where would need an empty-string-parameter.

Reply:
I am also not sure. But I think there are cases in which the API differentiates between an empty and a non-existent parameter, e.g. with the parameter ``pass`` in ``/validate/check``: The endpoint complains if ``pass`` is not supplied, but works fine if ``pass`` is empty (as seen in #1838)

Reply:
OK, I will change it to ``is not None``.

Reply:
done in e1e123d0

Reply:
thanks!